src/lib/utils/sendBeacon.spec.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ *
+ * stylelounge.de
+ *
+ * Copyright (C) SNM Style Net Media GmbH
+ * MIT Licensed
+ */
+import {default as send} from "./sendBeacon";
+import * as sinon from "sinon";
+import IManifest from "../types/IManifest";
+declare let global: any;
+describe("sendBeacon ", function () {
+    const manifest: IManifest = {
+        data: {
+            mock: "data",
+        },
+        id: 1,
+        url: "/some-mock-url",
+        verb: "POST",
+    };
+    beforeEach(function () {
+        global.navigator = sinon.stub(global, "navigator");
+        global.navigator.sendBeacon = sinon.stub(global.navigator, "sendBeacon");
+        global.Blob = sinon.stub(global, "Blob");
+        global.fetch = sinon.stub(global, "fetch");
+    });
+    afterEach(function () {
+        global.navigator.sendBeacon.restore();
+        global.navigator.restore();
+        global.fetch.restore();
+        global.Blob.restore();
+    });
+    it("should send request with beacon", async function () {
+        global.navigator.sendBeacon.resolves(true);
```

Comment:
sendBeacon doesn't return a promise but a boolean. This stub returns a promise that is resolved with true
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon